sources/packages/web/src/views/aest/student/CASSupplierInformation.vue

```diff
@@ -0,0 +1,157 @@
+                item.supplierProtected
+                  ? "Yes"
+                  : item.supplierProtected === false
+                  ? "No"
+                  : "-"
+              }}
+            </template>
+            <template #[`item.supplierStatus`]="{ item }">
+              <status-chip-supplier :status="item.supplierStatus" />
+            </template>
+            <template #[`item.isValid`]="{ item }">
+              {{ item.isValid ? "Yes" : "No" }}
+            </template>
+            <template #[`item.supplierSiteCode`]="{ item }">
+              {{ item.supplierSiteCode }}
+            </template>
+            <template #[`item.addressLine1`]="{ item }">
+              {{ item.addressLine1 || "-" }}
+            </template>
+            <template #[`item.siteStatus`]="{ item }">
+              {{ item.siteStatus || "-" }}
+            </template>
+            <template #[`item.siteProtected`]="{ item }">
+              {{ item.siteProtected || "-" }}
+            </template>
+          </v-data-table>
+        </toggle-content>
+      </content-group>
+      <edit-c-a-s-supplier-info-modal
+        ref="addCASSupplierModal"
+        :allowedRole="Role.AESTEditCASSupplierInfo"
+      />
+    </body-header-container>
+  </tab-container>
+</template>
+
+<script lang="ts">
+import { ref, watch, defineComponent } from "vue";
+import {
+  DEFAULT_PAGE_LIMIT,
+  Role,
+  CASSupplierInformationHeaders,
+} from "@/types";
+import { ModalDialog, useSnackBar, useFormatters } from "@/composables";
+import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
+import EditCASSupplierInfoModal from "@/components/aest/students/EditCASSupplierInfoModal.vue";
+import { CASSupplierInformation } from "@/types/contracts/student/CASSupplierInformation";
+import {
+  AddCASSupplierAPIInDTO,
+  CASSupplierInfoAPIOutDTO,
+} from "@/services/http/dto";
+import { CASSupplierService } from "@/services/CASSupplierService";
+import StatusChipSupplier from "@/components/generic/StatusChipSupplier.vue";
+
+export default defineComponent({
+  components: {
+    CheckPermissionRole,
+    EditCASSupplierInfoModal,
+    StatusChipSupplier,
+  },
+  props: {
+    studentId: {
+      type: Number,
+      required: true,
+    },
+  },
+  setup(props) {
+    const { dateOnlyLongString } = useFormatters();
+    const showModal = ref(false);
+    const casSuppliers = ref([] as CASSupplierInfoAPIOutDTO[]);
+    const addCASSupplierModal = ref(
+      {} as ModalDialog<AddCASSupplierAPIInDTO | boolean>,
+    );
+    const snackBar = useSnackBar();
+    const initialData = ref({ studentId: props.studentId });
+    const casSupplierResults = ref([] as CASSupplierInformation[]);
+    const loadCASSuppliers = async () => {
+      casSuppliers.value =
+        await CASSupplierService.shared.getSupplierInfoByStudentId(
+          props.studentId,
+        );
+    };
+    watch(() => props.studentId, loadCASSuppliers, { immediate: true });
```

Comment:
Minor but we can use `watchEffect` instead to simplify the code.
It is used in other places also.